crates/provider/src/kubernetes/client.rs

```diff
@@ -369,6 +386,40 @@ impl KubernetesClient {
         ))
     }
 
+    /// Create resources from yamls in `static-configs` directory
+    pub(super) async fn create_static_resource(
+        &self,
+        namespace: &str,
+        raw_manifest: &str,
+    ) -> Result<()> {
+        let tm: TypeMeta = serde_yaml::from_str(raw_manifest).map_err(|err| {
+            Error::from(anyhow!(
+                "error while extracting TypeMeta from manifest: {raw_manifest}: {err}"
+            ))
+        })?;
+        let gvk = GroupVersionKind::try_from(&tm).map_err(|err| {
+            Error::from(anyhow!(
+                "error while extracting GroupVersionKind from manifest: {raw_manifest}: {err}"
+            ))
+        })?;
+
+        let ar = ApiResource::from_gvk(&gvk);
+        let api: Api<DynamicObject> = Api::namespaced_with(self.inner.clone(), namespace, &ar);
+
+        api.create(
+            &PostParams::default(),
+            &serde_yaml::from_str(raw_manifest).unwrap(),
+        )
+        .await
+        .map_err(|err| {
+            Error::from(anyhow!(
+                "error while creating static-config {raw_manifest}: {err}"
+            ))
```

Comment:
Maybe we should write the manifest as Rust, this way, we benefit from typechecking and now if something doesn't works between kubernetes version upgrade due to API changes:

[LimitRange](https://docs.rs/k8s-openapi/latest/k8s_openapi/api/core/v1/struct.LimitRange.html)
[NetworkPolicy](https://docs.rs/k8s-openapi/latest/k8s_openapi/api/networking/v1/struct.NetworkPolicy.html)

Wdyt ?

Reply:
I think keep those manifest as `yaml` is ok, since most of the time are very static and if we need to change something is easily to review with @emamihe